src/providers/core_provider/mod.rs

```diff
@@ -71,41 +71,19 @@ pub struct CoreProviderBuilder {
 }
 
 impl CoreProviderBuilder {
-    pub fn new() -> std::io::Result<Self> {
-        let crate_version: Version = Version::from_str(version!()).or_else(|e| {
-            format_error!("Error parsing the crate version", e);
-            Err(Error::new(
-                ErrorKind::InvalidData,
-                "crate version number has invalid format",
-            ))
-        })?;
-
-        let provider_info = vec![ProviderInfo {
-            // Assigned UUID for this provider: 47049873-2a43-4845-9d72-831eab668784
-            uuid: Uuid::parse_str("47049873-2a43-4845-9d72-831eab668784").or_else(|_| Err(Error::new(
-                ErrorKind::InvalidData,
-                "provider UUID is invalid",
-            )))?,
-            description: String::from("Software provider that implements only administrative (i.e. no cryptographic) operations"),
-            vendor: String::new(),
-            version_maj: crate_version.major,
-            version_min: crate_version.minor,
-            version_rev: crate_version.patch,
-            id: ProviderID::Core,
-        }];
```

Comment:
Yeah, because in `build` I can just append the `CoreProvider`, so it's always last. Otherwise I'd have to always insert `ProviderInfo` structures in the 2nd to last place, bit more tedious.

Reply:
Makes sense! Would be nice as well for clients that the first provider is the "most priority" one and that the Core Provider is always the last one. Let's not forget to add that in the book!